sql/_36_guava/partition_table/cases/cbrd_25519.sql

```diff
@@ -0,0 +1,241 @@
+drop table if exists ta_range, ta_hash, ta_list, tb;
+create table ta_range (ca int, cb int, cc int, cd int, index i1 (cb, cc), primary key (ca, cd))
+partition by range (ca)
+(
+  partition p0 values less than (1),
+  partition p1 values less than (2),
+  partition p2 values less than (3),
+  partition p3 values less than (4),
+  partition p4 values less than (5),
+  partition p5 values less than (6),
+  partition p6 values less than (7),
+  partition p7 values less than (8),
+  partition p8 values less than (9),
+  partition p9 values less than MAXVALUE
+);
+create table ta_hash (ca int, cb int, cc int, cd int, index i1 (cb, cc), primary key (ca, cd)) partition by hash (ca) partitions 8;
+create table ta_list (ca int, cb int, cc int, cd int, index i1 (cb, cc), primary key (ca, cd))
+partition by list (ca)
+(
+  partition p0 values in (0, 1),
+  partition p1 values in (2, 3),
+  partition p2 values in (4, 5),
+  partition p3 values in (6, 7),
+  partition p4 values in (8, 9)
+);
+create table tb (ca int, cb int, cc int, cd int, index i1 (cb, cc), primary key (ca, cd));
+
+insert into tb
+  with recursive cte (n) as (
+      select 0
+      union all
+      select n + 1 from cte where n < 9
+    )
+  select a.n, b.n, c.n, rownum from cte a, cte b, cte c;
+insert into ta_range select * from tb;
+insert into ta_list select * from tb;
+insert into ta_hash select * from tb;
+
+update statistics on ta_range, ta_list, ta_hash, tb with fullscan;
+
+set trace on;
+
+-- START TEST
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. agl');
+
+select /*+ recompile */ count (*)
+from ta_range a
+using index none;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (range)');
+
+select /*+ recompile */ *
+from ta_range a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (hash)');
+
+select /*+ recompile */ *
+from ta_hash a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. all partitions (list)');
+
+select /*+ recompile */ *
+from ta_list a
+using index none
+order by cd desc
+limit 1;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (range)');
+
+select /*+ recompile */ count (*)
+from ta_range a
+where a.ca between 1 and 4;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (hash)');
+
+select /*+ recompile */ count (*)
+from ta_hash a
+where a.ca in (2, 3);
+show trace;
+
+select /*+ recompile */ count (*)
+from ta_hash a
+where a.ca >= 9;
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning (list)');
+
+select /*+ recompile */ count (*)
+from ta_list a
+where a.ca in (5, 6);
+show trace;
+
+do @i := @i + 1;
+evaluate concat ('####', lpad (@i, 3), '. pruning + limit');
+
+select /*+ recompile */ *
+from ta_range a
+order by cd
+limit 150, 10;
```

Comment:
ORDER BY 절이 추가되면 테스트 목적과 달라집니다.
아래처럼 변경해주세요.
```suggestion
select count(*)
from (
    select *
    from ta_range a
    limit 150, 10
  );
```

Reply:
리뷰 감사합니다. 모두 반영하였으니 확인 부탁드립니다.

Reply:
확인했습니다. 감사합니다.